go/vt/vtgate/engine/insert.go

```diff
@@ -40,322 +36,78 @@ import (
 
 var _ Primitive = (*Insert)(nil)
 
-type (
-	// Insert represents the instructions to perform an insert operation.
-	Insert struct {
-		// Opcode is the execution opcode.
-		Opcode InsertOpcode
+// Insert represents the instructions to perform an insert operation.
+type Insert struct {
+	*InsertCommon
```

Comment:
since `Insert` is already used by reference, I don't think we need the shared code to also be referenced